src/console_reporter.cc

```diff
@@ -36,6 +36,7 @@ namespace benchmark {
 bool ConsoleReporter::ReportContext(const Context& context) {
   name_field_width_ = context.name_field_width;
   printed_header_ = false;
+  prev_counters_.clear();
 
   PrintBasicContext(&GetErrorStream(), context);
 
```

Comment:
don't forget to update the WINDOWS build too. color_output_ is referenced in there.